dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -865,6 +882,149 @@ public DagDataSchedule exportProcessDagData(ProcessDefinition processDefinition)
+                    taskDefinition.setName(taskName);
+                    taskDefinition.setFlag(Flag.YES);
+                    SqlParameters sqlParameters = new SqlParameters();
+                    sqlParameters.setType(dataSource.getType().name());
+                    sqlParameters.setDatasource(dataSource.getId());
+                    sqlParameters.setSql(sql.substring(0, sql.length() - 1));
+                    sqlParameters.setSqlType(SqlType.NON_QUERY.ordinal());
+                    sqlParameters.setLocalParams(Collections.emptyList());
+                    taskDefinition.setTaskParams(JSONUtils.toJsonString(sqlParameters));
+                    taskDefinition.setCode(CodeGenerateUtils.getInstance().genCode());
+                    taskDefinition.setTaskType(TaskType.SQL.getDesc());
+                    taskDefinition.setFailRetryTimes(0);
+                    taskDefinition.setFailRetryInterval(0);
+                    taskDefinition.setTimeoutFlag(TimeoutFlag.CLOSE);
+                    taskDefinition.setWorkerGroup(DEFAULT_WORKER_GROUP);
+                    taskDefinition.setTaskPriority(Priority.MEDIUM);
+                    taskDefinition.setEnvironmentCode(-1);
+                    taskDefinition.setResourceIds("");
+
+                    taskDefinitionList.add(taskDefinition);
+                    taskNameToCode.put(taskDefinition.getName(), taskDefinition.getCode());
+                    taskNameToUpstream.put(taskDefinition.getName(), upstreams);
+                }
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            throw new ServiceException(Status.IMPORT_PROCESS_DEFINE_ERROR);
+        }
+
+        // build task relation
+        for (Map.Entry<String, Long> entry : taskNameToCode.entrySet()) {
+            List<String> upstreams = taskNameToUpstream.get(entry.getKey());
+            if (CollectionUtils.isEmpty(upstreams)
+                || (upstreams.size() == 1 && upstreams.contains("root") && !taskNameToCode.containsKey("root"))) {
+                ProcessTaskRelation processTaskRelation = new ProcessTaskRelation();
+                processTaskRelation.setPreTaskCode(0);
+                processTaskRelation.setPostTaskCode(entry.getValue());
+                processTaskRelation.setConditionType(ConditionType.NONE);
+                processTaskRelationList.add(processTaskRelation);
+                continue;
+            }
+            for (String upstream : upstreams) {
+                ProcessTaskRelation processTaskRelation = new ProcessTaskRelation();
+                processTaskRelation.setPreTaskCode(taskNameToCode.get(upstream));
+                processTaskRelation.setPostTaskCode(entry.getValue());
+                processTaskRelation.setConditionType(ConditionType.NONE);
+                processTaskRelationList.add(processTaskRelation);
+            }
+        }
+
+        checkAndImport(loginUser, projectCode, result, dagDataSchedule);
```

Comment:
I think there is an essential difference between this and import. It is not appropriate to use `checkAndImport`